Blender 3 and Cycles x

themasterlink opened the pull request:
This PR changes everything to blender 3.0.0 and Cycles X.

This means `set_samples` is removed and replaced with `set_max_amount_of_samples`. However, to control the noise one should now use the `set_noise_threshold` fct.

The docu was changed accordingly. 

The default values are the same as the default values used by blender, which should work for most users.

Fixes #360

themasterlink commented:
BlenderProcHelper result for run 1:

BlenderProcHelper has started, waiting for first results.

themasterlink commented:
BlenderProcHelper result for run 1:

BlenderProcHelper has started, waiting for first results.

themasterlink commented:
BlenderProcHelper result for run 60:

BlenderProcHelper has started, waiting for first results.

themasterlink commented:
BlenderProcHelper result for run 62:

BlenderProcHelper has started, waiting for first results.

themasterlink commented:
BlenderProcHelper result for run 63:

BlenderProcHelper has started, waiting for first results.

themasterlink commented:
BlenderProcHelper result for run 64:

BlenderProcHelper has started, waiting for first results.

themasterlink commented:
BlenderProcHelper result for run 65:

BlenderProcHelper has started, waiting for first results.

themasterlink commented:
BlenderProcHelper result for run 66:

BlenderProcHelper has started, waiting for first results.

themasterlink commented:
BlenderProcHelper result for run 67:

BlenderProcHelper has started, waiting for first results.

themasterlink commented:
BlenderProcHelper result for run 69:

BlenderProcHelper has started, waiting for first results.

themasterlink commented:
BlenderProcHelper result for run 72:

BlenderProcHelper has started, waiting for first results.

MartinSmeyer commented:
The changes look good to me, I suppose the errors are from the tests themselves.

themasterlink commented:
The error in the SUNCG tests seems like a bug, I will fix this.

themasterlink commented:
BlenderProcHelper result for run 80:

BlenderProcHelper has started, waiting for first results.

themasterlink commented:
@MartinSmeyer I fixed the bug in the SUNCG, was not really a bug in SUNCG and more a bug in the `clear_all_cp` of the `Struct` class. They changed the iterating behavior of keys when some get deleted, I fixed that.
